### ISSUE TITLE
.Net 8 won't allow multiple indexes on a single property

### ISSUE THREAD

**Issue by mrpmorris** [dotnet/efcore#32484]
The following code should result in a Db migration that adds two indexes to a SQL Server Db.

```C#
internal class DomainEventLogPersistenceConfigurationForSqlServer : IEntityTypeConfiguration<DomainEventLog>
{
    public void Configure(EntityTypeBuilder<DomainEventLog> builder)
    {
        // This is fast for finding rows with PublishedUtc = null
        builder
            .HasIndex(x => x.PublishedUtc)
            .HasDatabaseName("IX_DomainEventLogs_PublishedUtcDescending")
            .HasFilter($"{nameof(DomainEventLog.PublishedUtc)} is null");

        // This is fast for finding most recently published rows.
        // This is useful for event replay after a disaster recovery.
        // But is currently not working
        builder
            .HasIndex(x => x.PublishedUtc)
            .IsDescending()
            .HasDatabaseName("IX_DomainEventLogs_PublishedUtcDescending")
            .HasFilter($"{nameof(DomainEventLog.PublishedUtc)} is not null");
    }
}
```

However, it only adds the last index defined.

If I add only one, then add a migration, then add the second, then the next migration I add will first remove the previously created index.

**Comment by Quintinon** [comment]
Both of the indexes have the same database name, which is also probably causing issues.

**Comment by roji** [comment]
@Quintinon you're right - that's also going to create issues, but note that if different **model** names are used (as in my snippet above), they also serve as the database name by default, so there's no real reason to use HasDatabaseName for that case.

**Comment by mrpmorris** [comment]
@Quintinon That was something I accidentally broke later, I think :) They did have different names to start with and I couldn't create two.

@roji Thank you for your answer, I now have it doing exactly what I need it to!

I won't close this, just in case you want to leave it open as a reminder for docs to be updated. If you don't need it, I am satisfied with my care (film quote) :)

**Comment by mrpmorris** [comment]
@roji I found the relevant documentation which does tell me how to achieve this, but I have a recommendation.

The problem with the documentation is that it says I can give an index a name, but doesn't tell me why I would want to. I think once I know what it achieves then it is pretty obvious what it is for, but before that point it isn't.

I'll close this and add a docs issue.

Thanks again!

**Comment by roji** [comment]
> I found the relevant documentation which does tell me how to achieve this

@mrpmorris which documentation are you referring to here?

**Comment by roji** [comment]
Submitted #4581 to document this.
